### ISSUE TITLE
Allow disabling project file check when using `-p` flag

### ISSUE THREAD

**Issue by euclio** [palantir/tslint#3677]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.6.2
- __Running TSLint via__: (pick one) CLI / vim (Neomake)

#### TypeScript code being linted

N/A

with `tslint.json` configuration:

N/A

#### Actual behavior

When using tslint with the `-p` flag, the linter checks that the files passed on the command line are not excluded in the `tsconfig.json`. 

This is an issue for a number of reasons:

1. When using the JSON formatter, the "file is not included in the project" error is not JSON formatted, which causes any plugin that relies on JSON output to fail.
2. I'd like to be able to lint tests, etc. that are excluded from the project, but are still typescript files. When the linter is called automatically through vim, there's no way for the vim plugin to know if the file is actually a part of the project's `tsconfig.json` or not. If not, the linter fails and reports that the file is not included in the project.
3. The vim plugin providers as-you-type linting by writing the current buffer to a temporary file, and then passing this filename into the linter. Similarly to 2, the linting of this temporary file will fail if it's not included in `tsconfig.json`. Instead, the plugin has had to disable the as-you-type linting for typescript. (Another solution would be to allow linting through STDIN, reported at #1590).

#### Expected behavior

tslint allows linting files that are not part of the tsconfig project if they are explicitly passed on the command line, or at least provides a flag to disable this behavior.

**Comment by idmontie** [comment]
> I'd like to be able to lint tests, etc. that are excluded from the project, but are still typescript files. When the linter is called automatically through vim, there's no way for the vim plugin to know if the file is actually a part of the project's tsconfig.json or not. If not, the linter fails and reports that the file is not included in the project.

This is my usecase as well. When working in es6+babel, I usually have my test files as `*.test.js` next to the file I'm testing. When I moved to typescript, it looks like this isn't really the norm.

**Comment by alexander-alzate** [comment]
This issue is related to #1590, maybe reading from `stdin` would be a better approach.

**Comment by Glazy** [comment]
From what I can see there has been no progress made on this issue in a while, did you manage to find a workaround for this issue @euclio?

I have the exact same use-case as both @euclio and @idmontie.

**Comment by fabb** [comment]
Wouldn‘t it be possible to use the `-p` flag combined with a separate `tsconfig-tests` That imports the main `tsconfig`, but overwrites the include/exclude options?

**Comment by michaelw85** [comment]
For anyone stumbling upon this page, you can work around this issue as @fabb suggested by creating an additional config.

Here's an example of my setup:

```js
// Package.json
"tslint": "./node_modules/.bin/tslint -c tslint.json -p tslint-tsconfig.json",
```
```js
// tsconfig.json
  "include": [
    // include everything, also files you only want to be linted
    "src/**/*.ts",
    "test/**/*.ts"
  ],
  "exclude": [
     // Now exclude what we do not want in our project
     "test/**/*.ts"
  ],
```

```js
// tslint-tsconfig.json
{
  "extends": "./tsconfig",
  "exclude": [] // Remove all excludes so everything will be linted
}
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
